build/.mxchip-ci.yml

```diff
@@ -1,28 +1,58 @@
-# Copyright (c) Microsoft. All rights reserved.
-# Licensed under the MIT license. See LICENSE file in the project root for full license information.
 name: $(BuildID)_$(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
-resources:
-- repo: self
-  clean: true
-phases:
-- phase: iotz
-  queue:
-    name: devicelab-01
+
+jobs:
+- job: Cross_Compile
+  pool:
+    vmImage: ubuntu-16.04
   steps:
   - script: |
-      sudo apt-get update && apt-get install -y \
-      curl \
-      git \
-      python-software-properties \
-      build-essential \
-      pkg-config
-      sudo curl -sL https://deb.nodesource.com/setup_6.x | bash -
-      sudo apt-get install -y nodejs
+      . build_all/mbed5/setup.sh
     displayName: 'setup'
+
   - script: |
-      . build_all/mbed5/build.sh
+      # . build_all/mbed5/build.sh
+      git clone -b rajeev/latest_mbed https://github.com/massand/mxchip_az3166_firmware.git mxchip
+      # rsync -avz --existing ./ mxchip/mbed-iot-devkit-sdk/cores/arduino/azure-iot-sdk-c/
+
+      sudo npm install -g iotz
+      sudo iotz update
+
+      cd mxchip
+      sudo git checkout -- mbed-iot-devkit-sdk/cores/arduino/azure-iot-sdk-c/provisioning_client/adapters/hsm_client_riot.c
+      sudo git checkout -- mbed-iot-devkit-sdk/cores/arduino/azure-iot-sdk-c/provisioning_client/deps/RIoT/Reference/DICE/DiceSha256.c
+      sudo git checkout -- mbed-iot-devkit-sdk/cores/arduino/azure-iot-sdk-c/provisioning_client/deps/RIoT/Reference/RIoT/Core/RIoTCrypt/RiotDerEnc.c
+      sudo git checkout -- mbed-iot-devkit-sdk/cores/arduino/azure-iot-sdk-c/provisioning_client/deps/RIoT/Reference/RIoT/Core/RIoTCrypt/RiotEcc.c
+      sudo git checkout -- mbed-iot-devkit-sdk/cores/arduino/azure-iot-sdk-c/provisioning_client/deps/RIoT/Reference/RIoT/Core/RIoTCrypt/RiotSha256.c
+      sudo iotz init mbed
+      sudo iotz compile
+
+      cd ..
     displayName: 'build'
+
   - script: |
+      # Check that this is the right source directory
       ls
-      sudo cp ./mxchip/BUILD/AZ3166/GCC_ARM/mxchip.bin /media/john/AZ31661
+      cp mxchip/BUILD/AZ3166/GCC_ARM/mxchip.bin $(Build.ArtifactStagingDirectory)
+    displayName: "Copy Files into Artifact repo"
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifacts: drop'
+    inputs:
+      pathtoPublish: '$(Build.ArtifactStagingDirectory)'
+      artifactName: 'drop'
+      publishLocation: 'Container'
+    condition: succeeded()
+
+- job: Execute
+  dependsOn: Cross_Compile
+  pool:
+    name: $(AGENTPOOL)
+  steps:
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Build Artifacts'
+    inputs:
+      artifactName: 'drop'
+
+  - script: |
+      cp ../a/drop/mxchip.bin /media/newt/AZ3166
```

Comment:
are you doing anything with the mxchip once it's on the device? this code only copies the files from drop.

Reply:
Also it would be better to use '$(System.ArtifactsDirectory)' than assuming that the 'a' name and structure always hold true. 

```
cp $(System.ArtifactsDirectory)/drop/mxchip.bin /media/newt/AZ3166
```

Reply:
Correct, there is nothing done with the mxchip yet, doing e2e tests for it is a next step.

I'll modify that path to $(System.ArtifactsDirectory)', sounds better.